drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -521,6 +537,118 @@ func getPortworxServiceSpec(
 	return newService
 }
 
+func (c *portworxBasic) createAndMaintainPxSaTokenSecret(cluster *corev1.StorageCluster, ownerRef *metav1.OwnerReference) error {
+	if err := c.createTokenSecretIfNotExist(cluster, ownerRef); err != nil {
+		return err
+	}
+	if err := c.maintainTokenSecret(cluster, ownerRef); err != nil {
+		return fmt.Errorf("failed to maintain the token secret for px container. %v", err)
+	}
+	return nil
+}
+
+func (c *portworxBasic) createTokenSecretIfNotExist(cluster *corev1.StorageCluster, ownerRef *metav1.OwnerReference) error {
+	secret := &v1.Secret{}
+	err := c.k8sClient.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      pxutil.PortworxServiceAccountTokenSecretName,
+			Namespace: cluster.Namespace,
+		}, secret)
+	if err != nil {
+		if errors.IsNotFound(err) || len(secret.Data) == 0 || len(secret.Data[v1.ServiceAccountTokenKey]) == 0 {
+			token, err := generateToken(cluster)
+			if err != nil {
+				return err
+			}
+			rootCaCrt, _ := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")
+			curTime := time.Now()
+			tokenRefreshTimeBytes, err := curTime.Add(time.Duration(tokenExpirationSeconds / 2)).MarshalBinary()
+			if err != nil {
+				return fmt.Errorf("error marshalling current time to bytes. %v", err)
+			}
+			secret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
```

Comment:
we should add "created by operator" label/annotation. Check other objects that operator creates.

Reply:
I think `ownerReference` serves the purpose?

Reply:
Yes, but I thought we added the label/annotation as well. I have asked this question on https://github.com/libopenstorage/operator/pull/1585